src/Rinds/RindsSpecRunner.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Runs a single RINDS spec.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\Rinds;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+
+/**
+ * Runs a single RINDS spec.
+ */
+class RindsSpecRunner {
+	/**
+	 * Run the spec.
+	 *
+	 * @param object $spec The spec to run.
+	 */
+	public static function run_spec( $spec ) {
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+
+		// Create or update the note.
+		$existing_note_ids = $data_store->get_notes_with_name( $spec->slug );
```

Comment:
Should this ever return more than one note id? Seems like there should never be duplicates correct?

Reply:
I don't think there's actually anything that prevents duplicate note names. This is a pattern I've seen wherever we need to update a note.

Reply:
I was thinking about this as well and may provide a small refactor while working on #4283.  I'm not sure if there's ever a use case for allowing multiple notes with the same name.